src/main/java/com/echobox/api/linkedin/connection/v2/AssetsConnection.java

```diff
@@ -0,0 +1,136 @@
+  
+  private static final String ASSETS = "/assets";
+  private static final String ACTION_KEY = "action";
+  private static final String REGISTER_UPLOAD = "registerUpload";
+  private static final String COMPLETE_MULTIPART_UPLOAD = "completeMultiPartUpload";
+  
+  /**
+   * Initialise the assets connection
+   * @param linkedInClient the LinkedIn client
+   */
+  public AssetsConnection(LinkedInClient linkedInClient) {
+    super(linkedInClient);
+  }
+  
+  /**
+   * Register an upload to declare the upcoming upload
+   * @see <a href="https://docs.microsoft.com/en-us/linkedin/marketing/integrations/
+   * community-management/shares/vector-asset-api#register-an-upload">Register an Upload</a>
+   * @param registerUploadRequestBody the register upload request body
+   * @return the register upload response
+   */
+  public RegisterUpload registerUpload(RegisterUploadRequestBody registerUploadRequestBody) {
+    return linkedinClient.publish(ASSETS, RegisterUpload.class, registerUploadRequestBody,
+        Parameter.with(ACTION_KEY, REGISTER_UPLOAD));
+  }
+  
+  /**
+   * Upload the asset file
+   * @see <a href="https://docs.microsoft.com/en-us/linkedin/marketing/integrations/
+   * community-management/shares/vector-asset-api#upload-the-asset">Upload the Asset</a>
+   * @param uploadURL the uploadUrl from the register upload response
+   * @param headers the headers from the register upload response
+   * @param filename the file name of the file to be uploaded
+   * @param file the file to upload
+   * @return the map of headers from the request
+   * @throws IOException IOException
+   * @throws GeneralSecurityException GeneralSecurityException
+   */
+  public Map<String, String> uploadAsset(URL uploadURL, Map<String, String> headers,
+      String filename, File file) throws IOException, GeneralSecurityException {
+    InputStream videoInputStream = new FileInputStream(file);
+    byte[] bytes = new byte[(int) file.length()];
+    videoInputStream.read(bytes);
+    return uploadAsset(uploadURL, headers, filename, bytes);
+  }
+  
+  /**
+   * Upload the asset bytes, this should be used to upload each asset chunk
+   * @see <a href="https://docs.microsoft.com/en-us/linkedin/marketing/integrations/
+   * community-management/shares/vector-asset-api#upload-the-asset">Upload the Asset</a>
+   * @param uploadURL the uploadUrl from the register upload response
+   * @param headers the headers from the register upload response
+   * @param filename the file name of the file to be uploaded
+   * @param bytes the bytes to upload
+   * @return the map of headers from the request
+   * @throws IOException IOException
+   * @throws GeneralSecurityException GeneralSecurityException
+   */
+  public Map<String, String> uploadAsset(URL uploadURL, Map<String, String> headers,
+      String filename, byte[] bytes) throws IOException, GeneralSecurityException {
+    WebRequestor.Response response = new DefaultWebRequestor(null)
```

Comment:
We cannot pass any other headers (like LinkedIn authorization bearer token) when making a request to the assets URL (i.e. S3) so all headers and authorization must be removed and it might be easier just to create a new WebRequestor.
Perhaps it's better to not force to use a DefaultWebRequestor here but rather the client they passed